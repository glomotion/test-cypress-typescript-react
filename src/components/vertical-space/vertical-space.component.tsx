import React, { cloneElement, Children } from 'react';
import { css } from 'emotion';
import classNames from 'classnames';

import styles from './vertical-space.style.module.css';

export function VerticalSpace({
  children,
  top,
  bottom,
  both,
  responsiveBreakpoints,
  testId,
}: {
  children: any;
  top?: string;
  bottom?: string;
  both?: string;
  responsiveBreakpoints?: any[];
  testId?: string;
}) {
  return children && children.props ? (
    cloneElement(Children.only(children), {
      className: classNames(
        children.props.className,
        styles.verticalSpace,
      ),
      ...(typeof children.type === 'string' ? { 'data-testid': testId } : { testId }),
    })
  ) : (
    <span
      className={classNames(
        styles.verticalSpace,
      )}
      data-testid={testId}
    >
      {children}
    </span>
  );
}

VerticalSpace.defaultProps = {
  top: '',
  bottom: '',
  both: '',
  testId: '',
  responsiveBreakpoints: [],
};
