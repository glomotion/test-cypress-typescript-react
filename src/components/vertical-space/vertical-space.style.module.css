.verticalSpace {
  
  /* DEFINE BASE SIZING */
  margin-top: var(--marginTop);
  margin-bottom: var(--marginBottom);
}