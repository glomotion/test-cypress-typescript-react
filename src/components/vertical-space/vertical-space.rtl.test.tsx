/// <reference types="Jest" />

import React from 'react';
import { findByTestId, render, screen } from '@testing-library/react';

import { VerticalSpace } from './vertical-space.component';

describe('<VerticalSpace />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <VerticalSpace>
        <div>mooo</div>
      </VerticalSpace>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <VerticalSpace>
        <div>mooo</div>
      </VerticalSpace>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
