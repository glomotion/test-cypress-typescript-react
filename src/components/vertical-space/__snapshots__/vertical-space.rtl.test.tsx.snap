// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerticalSpace /> Should render and match the snapshot 1`] = `
<div
  class=""
  data-testid=""
>
  mooo
</div>
`;
